lib/bcdice/game_system/Ayabito.rb

```diff
@@ -0,0 +1,229 @@
+# frozen_string_literal: true
+
+require 'bcdice/command/parser'
+
+module BCDice
+  module GameSystem
+    class Ayabito < Base
+      # ゲームシステムの識別子
+      ID = "Ayabito"
+
+      # ゲームシステム名
+      NAME = "あやびと"
+
+      # ゲームシステム名の読みがな
+      SORT_KEY = "あやひと"
+
+      HELP_MESSAGE = <<~TEXT
+        ・判定コマンド(xAB±y@c>=z)
+          x：サイコロの数(10以上の場合9個振り、それ以降を成功数2として加算する)
+          ±y：成功数への補正(省略可)
+          c：クリティカル値(@ごと省略可。省略時は6)
+          z：目標値(妨害値など。>=ごと省略可)
+          (例) 4AB
+               11AB>=5
+               5AB+1
+               6AB@5>=3
+
+        ・各種表
+          感情表 ET
+          帝都東京シーン表 TST / 場面演出シーン表 BST
+          交流表 CET
+          ファンブル表 FT
+          封印期間表 LT
+      TEXT
+
+      def initialize(command)
+        super(command)
+        @sort_barabara_dice = true
+        @round_type = RoundType::CEIL
+      end
+
+      def eval_game_system_specific_command(command)
+        return check_action(command) || roll_tables(command, TABLES)
+      end
+
+      def check_action(command)
+        parser = Command::Parser.new("AB", round_type: RoundType::CEIL).has_prefix_number.enable_critical.restrict_cmp_op_to(nil, :>=)
+        parsed = parser.parse(command)
+        return nil if parsed.nil?
+
+        if parsed.prefix_number < 10
+          dice_cnt = parsed.prefix_number
+          over_modify = 0
+        else
+          dice_cnt = 9
+          over_modify = parsed.prefix_number - 9
+        end
+        modify = parsed.modify_number
+        critical_target = parsed.critical || 6
+        target = parsed.target_number
+
+        dice_arr = @randomizer.roll_barabara(dice_cnt, 6).sort
+        dice_str = dice_arr.join(",")
+        has_critical = (dice_arr.count { |x| x >= critical_target } > 0)
```

Comment:
```suggestion
        has_critical = dice_arr.any? { |x| x >= critical_target }
```
[`Array#any?`](https://docs.ruby-lang.org/ja/latest/method/Array/i/any=3f.html) を使うのが良さそうです。